lib/csv.rb

```diff
@@ -1309,7 +1309,7 @@ def self.open(filename, mode="r", **options)
   #
   def self.parse(*args, &block)
     csv = new(*args)
-    if block.nil?  # slurp contents, if no block is given
+    unless block_given? # slurp contents, if no block is given
```

Comment:
`unless else` is confused.
How about the following?

```ruby
return csv.each(&block) if block_given?

begin
  csv.read
ensure
  csv.close
end
```

Reply:
I'll try it!